src/main/scala/uk/gov/nationalarchives/Main.scala

```diff
@@ -0,0 +1,76 @@
+package uk.gov.nationalarchives
+
+import cats.effect._
+import fs2.Stream
+import io.circe.{Decoder, HCursor}
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import pureconfig._
+import pureconfig.generic.auto._
+import pureconfig.module.catseffect.syntax._
+import uk.gov.nationalarchives.dp.client.fs2.Fs2Client
+import uk.gov.nationalarchives.Message._
+import java.net.URI
+import java.nio.file
+import java.util.UUID
+import scala.concurrent.duration.DurationInt
+
+object Main extends IOApp {
+
+  case class Config(
+      preservicaUrl: String,
+      preservicaSecretName: String,
+      sqsQueueUrl: String,
+      repoDir: String,
+      workDir: String,
+      proxyUrl: Option[URI]
+  )
+
+  private def sqsClient(config: Config): DASQSClient[IO] =
+    config.proxyUrl
+      .map(proxy => DASQSClient[IO](proxy))
+      .getOrElse(DASQSClient[IO]())
+
+  implicit val decoder: Decoder[Option[Message]] = (c: HCursor) =>
+    for {
+      ref <- c.downField("id").as[String]
+    } yield {
+      val idAndType = ref.split(":")
+      val id = UUID.fromString(idAndType.last)
+      val entityType = idAndType.head
+      entityType match {
+        case "io" => Option(InformationObjectMessage(id))
+        case "co" => Option(ContentObjectMessage(id))
+        case _    => None
```

Comment:
Do you think it's worth turning this into `case "so" => None` and then adding a catch all case after it in case something weird comes in e.g. We/Preservica do some accidental update?

Reply:
What would the catch all case do though, wouldn't that just return `None` as well? We control the entity event generator so any Preservica weirdness can be handled up there.